src/base/utils/net.cpp

```diff
@@ -38,5 +40,44 @@ namespace Utils
         {
             return !QHostAddress(ip).isNull();
         }
+
+        bool isValidSubnet(const QPair<QHostAddress, int> &subnet)
+        {
+            return subnet != qMakePair(QHostAddress(), -1);
```

Comment:
In the qt5 [source](https://code.woboq.org/qt5/qtbase/src/network/kernel/qhostaddress.cpp.html#1108) `parseSubnet` returns `qMakePair(QHostAddress(), -1)` for invalid subnets.